src/commands/new.rs

```diff
@@ -272,7 +282,15 @@ impl NewCommand {
         })
     }
 
+    fn is_bin(&self) -> bool {
+        self.bin || !self.lib
+    }
+
     fn generate_source(&self, target: &Option<RegistryPackageResolution>) -> Result<Cow<str>> {
+        if self.is_bin() {
+            return Ok("".into());
```

Comment:
Would it make sense for this string to contain a `fn main() { println!("Hello, World!"); }`?

Reply:
Currently the underlying `cargo new` populates that for us; this just prevents `cargo component new` from overwriting that.

Reply:
Ah, makes sense. Could you add a comment mentioning that?

Reply:
Will do.